provides.py

```diff
@@ -55,6 +59,10 @@ def flush_departed(self):
         do any cleanup logic required. """
         conv = self.conversation()
         conv.remove_state('{relation_name}.departed')
+        all_creds = db.get('creds')
+        for user, cred in list(all_creds.items()):
+            if cred['scope'] == conv.scope:
+                all_creds.pop(user)
```

Comment:
This is missing a `db.set` to persist the removal.

Reply:
thank you